assets/js/modules/pagespeed-insights/components/DashboardPageSpeedCTA.js

```diff
@@ -6,23 +24,28 @@ import { __ } from '@wordpress/i18n';
 /**
  * Internal dependencies
  */
+import Data from 'googlesitekit-data';
 import {
 	activateOrDeactivateModule,
 	getReAuthURL,
 	showErrorNotification,
-	getModulesData,
 } from '../../../util';
 import CTA from '../../../components/notifications/cta';
 import data from '../../../components/data';
 import GenericError from '../../../components/notifications/generic-error';
+import { STORE_NAME as MODULES_STORE } from '../../../googlesitekit/modules/datastore/constants';
+import { STORE_NAME as USER_STORE } from '../../../googlesitekit/datastore/user/constants';
+const { useSelect } = Data;
 
-const PageSpeedInsightsCTA = () => {
-	const {
-		active,
-		setupComplete,
-	} = getModulesData()[ 'pagespeed-insights' ];
+function DashboardPageSpeedCTA() {
+	const pagespeedInsightsModule = useSelect( ( select ) => select( MODULES_STORE ).getModule( 'pagespeed-insights' ) );
+	const canManageOptions = useSelect( ( select ) => select( USER_STORE ).hasCapability( 'googlesitekit_manage_options' ) );
+
+	if ( ! pagespeedInsightsModule ) {
+		return null;
+	}
 
-	const { canManageOptions } = global._googlesitekitLegacyData.permissions;
+	const { active, setupComplete } = pagespeedInsightsModule;
 
 	if ( ! canManageOptions && ! setupComplete ) {
```

Comment:
Shouldn't this be changed to `! canManageOptions || active && setupComplete`? If the user cannot manage options, we shouldn't show this, but if the module is already active and set up, we shouldn't show this either.

Reply:
@felixarntz i am not sure why we need to add `active` check here. That's how it works in the master right now:

https://github.com/google/site-kit-wp/blob/7f88b98c775446d2dcda7aafa8acde6d9a70bb23/assets/js/modules/pagespeed-insights/components/dashboard/PageSpeedInsightsCTA.js#L27

Reply:
@eugene-manuilov That's somewhat incorrect that it's there in `master`. It's also irrelevant though, since that component is only ever rendered if `! active || ! setupComplete`, which covers with what I suggested above. So we can:
* Either just use `! canManageOptions`, relying on this component only being used when `! active || ! setupComplete`.
* Or use `! canManageOptions || active && setupComplete`, which ensures the component itself never displays the CTA when the CTA would be irrelevant.

Reply:
Ok, updated it to be `! canManageOptions || ( active && setupComplete )`.